src/app/modals/modals.component.ts

```diff
@@ -36,50 +35,31 @@ import { MdDialogRef } from '@angular/material';
 })
 export class ModalsComponent implements DoCheck, OnInit {
 
-  // @ViewChild('staticModal')
-  // public staticModal: ModalDirective;
-
   @ViewChild('modalContainer', { read: ViewContainerRef })
   private modalContainer: ViewContainerRef;
 
   modal: ComponentRef<Component>;
   hasScrollY: boolean = false;
-  syncPercentage: number = 0;
-  syncString: string;
   closeOnEscape: boolean = true;
   enableClose: boolean;
 
   private logger: any = Log.create('modals.component');
 
-  constructor (
-    private _element: ElementRef,
-    private _resolver: ComponentFactoryResolver,
-    private _modalService: ModalsService,
-    public _dialogRef: MdDialogRef<ModalsComponent>
-  ) {
-    // update new modal, open it
-    this._modalService.getMessage().subscribe(
-      message => {
-        if (message.close) {
-          this.close();
-        } else {
-          this.open(message.modal, message.data);
-        }
-      }
-    );
-
-    // update progress bar blockstatus
-    this._modalService.getProgress().subscribe(
-      progress => this.updateProgress(<number>progress)
-    );
+  constructor(private _element: ElementRef,
+              private _resolver: ComponentFactoryResolver,
+              public _dialogRef: MdDialogRef<ModalsComponent>,
+              private state: StateService) {
   }
 
   ngOnInit() {
-    this.enableClose = this._modalService.enableClose;
+    this.state.observe('model:fullWidth:enableClose')
+      .subscribe(
```

Comment:
this should be `status => this.enableClose = status);`
we can get rid of the brackets please